tests/app_auth_test.py

```diff
@@ -309,3 +309,69 @@ def test_authnz_token_reissue(redis_client: redis.Redis) -> None:
     assert session
     assert session.token == new_token
     assert session.user == "some-user@example.com"
+
+
+def test_authnz_token_reissue_internal(redis_client: redis.Redis) -> None:
+    """Test requesting token reissuance to an internal audience."""
+    keypair = RSAKeyPair()
+    token = create_test_token(keypair, ["admin"])
+    session_secret = os.urandom(16)
+    app = create_test_app(keypair, session_secret)
+
+    with app.test_client() as client:
+        with patch("jwt_authorizer.authnz.get_key_as_pem") as get_key_as_pem:
+            get_key_as_pem.return_value = keypair.public_key_as_pem()
+            r = client.get(
+                "/auth?capability=exec:admin&audience=https://example.com/api",
+                headers={"Authorization": f"Bearer {token}"},
+            )
+            assert get_key_as_pem.call_args_list == [
+                call("https://test.example.com/", "some-kid")
+            ]
+
+    assert r.status_code == 200
+    new_token = r.headers["X-Auth-Request-Token"]
+    assert token != new_token
+    ticket = Ticket.from_str(
+        "oauth2_proxy", r.headers["X-Auth-Request-Token-Ticket"]
+    )
+
+    assert jwt.get_unverified_header(new_token) == {
+        "alg": config.ALGORITHM,
+        "typ": "JWT",
+        "kid": "some-kid",
+    }
+
+    decoded_token = jwt.decode(
+        new_token,
+        keypair.public_key_as_pem(),
+        algorithms=config.ALGORITHM,
+        audience="https://example.com/api",
+    )
+    assert decoded_token == {
+        "act": {
+            "aud": "https://example.com/",
+            "iss": "https://test.example.com/",
+            "jti": "some-unique-id",
+        },
+        "aud": "https://example.com/api",
+        "email": "some-user@example.com",
+        "exp": ANY,
+        "iat": ANY,
+        "isMemberOf": [{"name": "admin"}],
+        "iss": "https://test.example.com/",
+        "jti": ticket.as_handle("oauth2_proxy"),
+        "sub": "some-user",
+        "uid": "some-user",
+        "uidNumber": "1000",
+    }
+    now = time.time()
+    expected_exp = now + app.config["OAUTH2_JWT_EXP"] * 60
+    assert expected_exp - 5 <= decoded_token["exp"] <= expected_exp + 5
+    assert now - 5 <= decoded_token["iat"] <= now + 5
+
+    session_store = SessionStore("oauth2_proxy", session_secret, redis_client)
+    session = session_store.get_session(ticket)
+    assert session
+    assert session.token == new_token
+    assert session.user == "some-user@example.com"
```

Comment:
I honestly don't think I understand what this reissuance is, but the tests look good!